source/connection-monitoring-and-pooling/tests/README.md

```diff
@@ -21,6 +21,74 @@ The following tests have not yet been automated, but MUST still be tested:
 5. When a check out attempt fails because connection set up throws an error, assert that a ConnectionCheckOutFailedEvent
     with reason="connectionError" is emitted.
 
+### Pending Response
+
+Due to the complexity of managing a proxy layer, the following qualifying tests should only be run for non-ssl,
+non-auth, and non-compression connections.
+
+#### Recover Partially Read Response
+
+This test verifies that if only part of a response was read before the timeout, the driver can drain the rest of the
+response and reuse the connection for the next operation.
```

Comment:
This doesn't match the actual contents of the test, so either the description or the test needs to be updated.

But if its the description that is inaccurate - with `sendAll: true` and the events (only one pending read started + finished pair), the full response will be read in the first operation.  Don't we have coverage for this scenario from our unified tests?

Reply:
We are using proxy server here to control when/how server response arrived to the client side. So in step 2 in addition to the regular insert command payload, we have to add additional `proxyTest` property, which instruct proxy to emulate timeout. It works as following:
1) it send request to the server
2) based on the `sendBytes` parameter it stream requested amount of bytes instantly
3) it sleeps for `delayMS`
4) it streams rest of the response.


I think it makes sense to add this explanation to the test summary.

Reply:
Okay, maybe my confusion stems from the interaction between `sendBytes` and `sendAll`.  Does `sendAll` being enabled not mean that the proxy will forward the full response back to the client?

Reply:
Yes. I've updated steps with sample of payload, it might help to understand the idea.